src/System.Reflection.Metadata/src/System/Reflection/Metadata/Handles.cs

```diff
@@ -652,6 +652,18 @@ public override int GetHashCode()
         {
             return left._rowId != right._rowId;
         }
+
+        /// <summary>
+        /// Returns a handle to <see cref="MethodDebugInformation"/> corresponding to this handle.
+        /// </summary>
+        /// <remarks>
+        /// The resulting handle is only valid within the context of a <see cref="MetadataReader"/> open on the Portable PDB blob,
+        /// which in case of standalone PDB file is a different reader than the one containing this method definition.
+        /// </remarks>
+        public MethodDebugInformationHandle ToDebugInformationHandle()
```

Comment:
I don't think we need to worry about the coupling here. By design all the handles are known to the metadata reader and I can't imagine splitting them up. A static From would be less discoverable and an extension would create a new type just for that, which would add clutter IMHO.